File: Algorithms/29. Divide Two Integers/golang/solution.go
import "math"

func divide(dividend int, divisor int) int {
	a := dividend / divisor
	if a <= int(math.Pow(2.0, 31.0)-1) && a >= int(math.Pow(-2.0, 31.0)) {
		return a
	} else {
		return a - 1
	}
}

File: Algorithms/29. Divide Two Integers/javascript/solution.js
/**
 * @param {number} dividend
 * @param {number} divisor
 * @return {number}
 */
var divide = function (dividend, divisor) {
  let res = Math.trunc(dividend / divisor);
  const max = Math.pow(2, 31) - 1;
  const min = Math.pow(-2, 31);

  if (res > max) {
    return max;
  } else if (res < min) {
    return min;
  }

  return res;
};
